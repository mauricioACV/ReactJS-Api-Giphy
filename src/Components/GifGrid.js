import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    //Usar mi Custom Hook
    const {data:images, loading} = useFetchGifs(category);
    return (
        <>
        <h3 className="animate__animated animate__fadeIn">{category}</h3>
        {/* {loading? 'Cargando...' : 'Cargado...'} */}
        {loading && <p className="animate__animated animate__flash">Loading</p>}
        <div className="card-grid">
            {
                // images.map(img => (
                //     <li key={img.id}>{img.title}</li>
                // ))
                images.map(img => (
                    <GifGridItem
                        key={img.id}
                        // img={img}
                        {...img}
                    />
                ))
            }
        </div>
        </>
    )
}
