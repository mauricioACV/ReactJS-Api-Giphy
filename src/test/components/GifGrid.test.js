import React from 'react'
import {shallow} from 'enzyme';
import { GifGrid } from '../../Components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
//Lo que haremos con este jest.mock, es fingir cualquier llamada al path que pasamos entre paréntesis, y suponer o controlar la
//información que nos retornaría
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {

    const category = 'One Punch';

    test('debe de mostrar el componente correctamente', () => {
        //Mock estado inicial de useFetchGifs
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imágenes useFetchGifs', () => {

        //usaremos un mock simulando que tenemos data que nos haya regresado nuestro custom hook
        const gifs = [{
            id: '123',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);

        //Es necesario el snapshot para evaluar los elementos que contiene
        expect(wrapper).toMatchSnapshot();
        //La prueba evalua que no exista el elemento parrafo (p) ya que no debería estar porque estamos simulando con el mock que ya renderizamos
        //resultados de gifs y el parrafo solo aparece cuando se esta realizando la consulta a la API
        expect(wrapper.find('p').exists()).toBe(false);
        //Evaluo que los elementos renderizados sean los mismos que contiene la data:gifs en el mock
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    });
    
    

});