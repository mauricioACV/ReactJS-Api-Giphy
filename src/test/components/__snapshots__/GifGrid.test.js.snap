// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
